lib/errors.ts

```diff
@@ -122,6 +122,22 @@ const messages: {
 
 	EACCES: (e) => messages.EPERM(e),
 
+	BalenaSettingsPermissionError: () => stripIndent`
+		You don't have sufficient privileges to access balena settings.
```

Comment:
I think this sentence would be better reading something like:
```
Error reading data directory: "${dataDirectory}"
```
... where `dataDirectory` might be obtained with something like:
```
const settings = await import('balena-settings-client');
const dataDirectory = settings.get('dataDirectory') as string;
```
Similar to how it is being done in [events.ts L67](https://github.com/balena-io/balena-cli/blob/v12.27.4/lib/events.ts#L67). Plus some error handling in case obtaining the value of `dataDirectory` fails as well, perhaps for the same reasons as reading the data directory fails (ownership, permissions). In case an error happens attempting to obtain the value of `dataDirectory`, a default value could be used.

I think we should avoid the word "settings" because it is similar to "configuration", and the CLI uses a configuration file which is typically `~/.balenarc.yml` -- see [more details in the FAQ](https://github.com/balena-io/balena-cli/blob/master/TROUBLESHOOTING.md#where-is-the-balena-clis-configuration-file-located). The configuration file and the data directory are different things that may both/either fail independently. The error handling in this PR regards the data directory, not the configuration file.

The location of the data directory is typically `~/.balena` but can be changed using the configuration file (typically `~/.balenarc.yml`) or by using the BALENARC_DATA_DIRECTORY environment variable. More details at:
* https://github.com/balena-io/balena-cli/blob/master/TROUBLESHOOTING.md#how-do-i-make-the-balena-cli-persist-data-in-another-directory
* https://www.balena.io/docs/learn/more/masterclasses/advanced-cli/#13-separate-environment-configurations

By the way, we should test that the `BalenaSettingsPermissionError` is still raised as expected if the data directory is modified using the configuration file or env var.

Reply:
I'm going to work on this soon.